package/MDAnalysis/analysis/pca.py

```diff
@@ -0,0 +1,224 @@
+        Find the principal components of selected atoms in a subset of the
+        trajectory.
+    transform(traj=None, n_components=None)
+        Take the original trajectory and project it onto the principal
+        components.
+    inverse_tranform(pca_space)
+        Take a pca_space and map it back onto the trajectory used to create it.
+    """
+
+    def __init__(self, atomgroup, demean=True, reference=None,
+                 n_components=None, **kwargs):
+        """
+        Parameters
+        ----------
+        atomgroup: MDAnalysis atomgroup
+            AtomGroup to be used for PCA.
+        n_components : int, optional
+            The number of principal components to be saved, default saves
+            all principal components, Default: -1
+        start : int, optional
+            First frame of trajectory to use for generation
+            of covariance matrix, Default: None
+        stop : int, optional
+            Last frame of trajectory to use for generation
+            of covariance matrix, Default: None
+        step : int, optional
+            Step between frames of trajectory to use for generation
+            of covariance matrix, Default: None
+        """
+        super(PCA, self).__init__(atomgroup.universe.trajectory,
+                                  **kwargs)
+        self._u = atomgroup.universe
+        self.start = kwargs.get('start')
+        self.stop = kwargs.get('stop')
+        self.step = kwargs.get('step')
+
+        if demean:
+            if frame is None:
+                self.reference = self._u
+            else:
+                # change u to traj number for fitting
+                self._u.trajectory[frame]
+                self.reference = self._u
+            aligned = align.AlignTraj(self.reference, self._u).run()
+            # need help here
+            self._u = mda.Universe(aligned.filename,)
+            atomgroup.universe = self._u
+
+        self._atoms = atomgroup
+        self.n_components = n_components
+        self._n_atoms = self._atoms.n_atoms
+
+    def _prepare(self):
+        n_dim = self._n_atoms * 3
+        self.cov = np.zeros((n_dim, n_dim))
+        self.mean = np.zeros(n_dim,)
+
+        for i, ts in enumerate(self._u.trajectory[self.start:self.stop:self.step]):
+            x = self._atoms.positions.ravel()
+            self.mean += x
+
+        self.mean /= self.n_frames
+
+    def _single_frame(self):
+        x = self._atoms.positions.ravel()
+        x -= self.mean
+        self.cov += np.dot(x[:, np.newaxis], x[:, np.newaxis].T)
+
+    def _conclude(self):
+        self.cov /= self.n_frames - 1
+        e_vals, e_vects = np.linalg.eig(self.cov)
+        sort_idx = np.argsort(e_vals)[::-1]
+        self.variance = e_vals[sort_idx]
+        self.p_components = e_vects[sort_idx]
+        self.cumulated_variance = (np.cumsum(self.variance) /
+                                   np.sum(self.variance))
+
+    def transform(self, atomgroup, n_components=None):
```

Comment:
This can't just take an atomgroup. You have to accept a universe and a new iteration slice as well.
It is legitimate to transform the coordinates of an Atomgroup of another universe with the same PCA object see.

```python
u1 = mda.Universe(sim1.pdb)
u2 = mda.Universe(sim2.pdb)

pca = PCA(u1).run()
pca_space1 = pca.transform(u1)
u2_in_pca_space1 = pca.transform(u2, start=3, stop=30)
```

This happens if you have several simulations of a protein and want to know how the second simulation looks if you project in into the PCA space of the first. Another use-case you calculate the PCA on a bunch of crystal structures of a protein and then project a simulation into that PCA.

You can take the trajectory argument from the atomgroup that you get 

```python
reader = ag.universe.trajectory
```